proposals/005_New_Security_Definitions.md

```diff
@@ -0,0 +1,531 @@
+| Review Manager | TBD |
+| Status | Proposal |
+| Implementations | N/A |
+| Issues | [{issueid}](https://github.com/OAI/OpenAPI-Specification/issues/{Issueid}) |
+| Previous Revisions | N/A |
+
+## Change Log
+
+| Date | Responsible Party | Description |
+| 2021/05/13 | Jeremy Whitlock | Initial proposal |
+
+## Introduction
+
+This purpose of this proposal is to rethink how
+[Security Schemes](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#security-scheme-object) are
+defined in OpenAPI.  This proposal would separate the credential location(s) from the Security Scheme _type_ and would
+allow for complete control over dictating where the credential(s) are provided for a Security Scheme.  This proposal
+will add a generic way to retrieve a credential from anywhere in a request, and allow any Security Scheme to dictate
+where its supported credential(s) are instead of relying on OpenAPI-supported conventions/opinions.  This proposal also
+separates the credential location(s) details from the other configurations associated with the Security Scheme
+_(like JWKS details for JWT)_.
+
+**Note:** The proposal isn't to remove `type` from the Security Scheme object but to allow for more control over where
+the credential(s) are provided for a Security Scheme.  This proposal would still have reasonable defaults for the
+already-supported `type` values and will be documented as such.
+
+## Motivation
+
+[Security Schemes](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#security-scheme-object)
+_(formerly known as "Security Definitions")_ provide a way to describe the security scheme(s) used by your API.
+Unfortunately, the _types_ supported by OpenAPI are not only limited but they often tie the credential location to the
+Security Scheme type.  This proposal will add a generic way to define where a credential is and its format so that the
+consumer can have complete control over where credentials are used in their API.  Not only would this allow consumers to
+specify their own Security Schemes without OpenAPI having to be the gatekeeper, but it also allows a level of
+flexibility not available now.  There have also been extra Security Scheme properties that have been _tacked on_ to
+support the growing needs of consumers, which would be alleviated _(or moved in this case)_.
+
+For example, if a consumer wants to use a JSON Web Key _(JWT)_ in their API, OpenAPI will allow this but only under
+certain circumstances:
+
+* `type: apiKey`: This will allow you to provide your JWT anywhere you would like, but it could be confusing to overload
+  `type: apiKey` since you are describing a Security Scheme using a JWT and not an API Key
+* `type: http`: This will only work if your JWT is in a header and in a supported
+  [HTTP Auth](https://datatracker.ietf.org/doc/html/rfc7235#section-5.1) format
+
+OpenAPI being so opinionated about these kinds things does not serve any real benefit to anyone involved, and this
```

Comment:
> OpenAPI being so opinionated about these kinds things does not serve any real benefit to anyone involved

Unsure this is totally true. Tooling authors may appreciate the limited set of possibilities they have to support?

Reply:
It's subjective for sure but is focusing on the tooling authors more important than the consumers of OpenAPI?  I think with a clean implementation, it shouldn't be too hard to support something like this as a fellow tooling author to another.